### PR TITLE
Fixes #90. Add Intern unit tests for isReportableUrl.

### LINKED ISSUE #90: Add Intern unit tests for isReportableUrl
https://github.com/webcompat/webcompat-reporter-extensions/blob/master/shared/base.js#L16

This will involve a little refactoring:

1. Pull out the isReportableURL into either its own module, or at least in the outer scope
2. Write a unit test that imports it
3. Make sure it gets run from the intern command (and passes!)

### PR DESCRIPTION
**Pull request by laghee**
r? @miketaylr

Lots of messy committing! (And I might have missed something in my delirium, just let me know.)

Edit: OK, more messy commit snarls, but I think I got all the leftover comments & the very, very sad comma.

### CODE REVIEW

**Comment by miketaylr** [comment]
Awesome!

Can you check out the error message from Travis? 

https://travis-ci.org/webcompat/webcompat-reporter-extensions/builds/380409861#L465

(I should be able to review this tomorrow morning)

File: package.json [hunk 1]
```diff
@@ -40,9 +40,11 @@
...
     "shelljs": "^0.8.1",
     "shx": "^0.2.2",
+    "systemjs": "^0.21.0",
+    "systemjs-plugin-babel": "0.0.25",
     "web-ext": "^2.4.0",
     "webpack": "^4.1.1",
     "webpack-cli": "^2.0.10"
   },
-  "license": "MPL-2.0"
+  "license": "MPL-2.0",
```
Comment (miketaylr) [inline package.json]: Ah, I guess a trailing comma makes the JSON parser sad.

File: tests/lib/helpers.js [hunk 2]
```diff
@@ -2,11 +2,14 @@
  * http://creativecommons.org/publicdomain/zero/1.0/ */
 
 const webpack = require("webpack");
+// import webpack from "webpack";
```
Comment (miketaylr) [inline tests/lib/helpers.js]: Is this a leftover comment? If so, can we remove?

File: tests/lib/helpers.js [hunk 3]
```diff
@@ -2,11 +2,14 @@
...
 
 /*
   Returns a promise that resolves once webpack has compiled all the addon
   configs have been built.
 */
+// export default function
```
Comment (miketaylr) [inline tests/lib/helpers.js]: Leftover comment?

File: tests/test-manifest.js [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
 
+import helpers from "./lib/helpers.js";
+import shell from "shelljs";
+
 const { registerSuite } = intern.getInterface("object");
 const { assert } = intern.getPlugin("chai");
 
-const helpers = require("./lib/helpers");
-const shell = require("shelljs");
+// const helpers = require("./lib/helpers");
```
Comment (miketaylr) [inline tests/test-manifest.js]: Leftover comments?

File: package.json (hunk 1 above)
Comment (laghee) [reply package.json]: Yep. Apparently I went spastic before pushing ... Now I'm having all kinds of merge issues before it'll let me push a fix. I'll address the other stuff, too, while I'm at it!

File: .babelrc [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+  "plugins": ["transform-es2015-modules-commonjs"]
+}
```
Comment (miketaylr) [inline .babelrc]: Can we add a newline here?

screen shot 2018-05-18 at 10 00 16 am

(and i'm pretty inconsistent about this myself... i should configure my editor to force this)

File: intern.json [hunk 6]
```diff
@@ -1,4 +1,13 @@
...
+  "description": "Default test suites",
+  "debug": true,
+  "filterErrorStack": true,
   "suites": "tests/**/*.js",
-  "environments": ["node"]
-}
\ No newline at end of file
+  "node": {
+    "plugins":
+      "node_modules/babel-register/lib/node.js"
```
Comment (miketaylr) [inline intern.json]: This is cool... I never really took the time to learn about this: https://github.com/theintern/intern/blob/master/docs/configuration.md#configuring-plugins

File: tests/test-checkurl-import.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+      );
+      assert.isTrue(
+        isReportableURL(httpsUrl),
+        "'https:' is a reportable URL protocol"
+      );
+    },
+    "calling isReportableURL with invalid protocols will not return true"() {
+      // Verify that calling isReportableURL with invalid protocol does not return true.
+      let guacUrl = "guacamole://yumavocados.com";
+      let ftpUrl = "ftp://downloadfreetacos.com";
```
Comment (miketaylr) [inline tests/test-checkurl-import.js]: 🌮

**Comment by miketaylr** [comment]
One tiny tip for rebasing against master so you don't end up with merge commits.

Since you're doing this from your fork, you need to make sure you have an `upstream` remote:

`git remote -v` will tell you that, and if you don't (but I assume you do?), https://help.github.com/articles/configuring-a-remote-for-a-fork/ has some help on that.

So the first step is to sync up remotes:

`git remote update`

This won't do anything destructive, it'll just make sure it knows about the latest changes on the server.

Then you want to do the rebase, which essentially means to put your local patches on top of any new stuff on master.

`git pull --rebase upstream master`

This is saying, pull in the latest stuff from the master branch on upstream remote, but as a rebase.

Assuming that works, you'll see something like "replaying your commits on top of blah".

But at this point, your local branch will be different from whatever you've already pushed to GitHub, so if you try to `git push` you'll get rejected. That's OK. It just means you have to force push:

`git push -f`

Just don't get too comfortable force pushing. :) It's always good to make sure you're not on master, for example. :))

Once you've done that, you've got the latest commits, and no merge commits, etc.

**Comment by laghee** [comment]
That makes sense. I've always been suspicious of `-f` and avoided it, so now I understand why merging was always wonky.

**Comment by miketaylr** [comment]
Thank you!

**Pull request merged**